test/colony-network-mining.js

```diff
@@ -668,40 +696,34 @@ contract("ColonyNetworkMining", accounts => {
       await giveUserCLNYTokensAndStake(colonyNetwork, MAIN_ACCOUNT, "1000000000000000000");
 
       const addr = await colonyNetwork.getReputationMiningCycle(true);
-      await forwardTime(3600, this);
+      await forwardTime(3000, this);
       const repCycle = await IReputationMiningCycle.at(addr);
-      await repCycle.submitRootHash("0x12345678", 10, 1);
-      await repCycle.submitRootHash("0x12345678", 10, 2);
-      await repCycle.submitRootHash("0x12345678", 10, 3);
-      await repCycle.submitRootHash("0x12345678", 10, 4);
-      await repCycle.submitRootHash("0x12345678", 10, 5);
-      await repCycle.submitRootHash("0x12345678", 10, 6);
-      await repCycle.submitRootHash("0x12345678", 10, 7);
-      await repCycle.submitRootHash("0x12345678", 10, 8);
-      await repCycle.submitRootHash("0x12345678", 10, 9);
-      await repCycle.submitRootHash("0x12345678", 10, 10);
-      await repCycle.submitRootHash("0x12345678", 10, 11);
-      await repCycle.submitRootHash("0x12345678", 10, 12);
-      await checkErrorRevert(repCycle.submitRootHash("0x12345678", 10, 13), "colony-reputation-mining-max-number-miners-reached");
+      let entryNumber = await getValidEntryNumber(colonyNetwork, MAIN_ACCOUNT, "0x12345678", 1);
+      for (let i = 1; i <= 12; i += 1) {
+        await repCycle.submitRootHash("0x12345678", 10, entryNumber); // eslint-disable-line no-await-in-loop
+        entryNumber = await getValidEntryNumber(colonyNetwork, MAIN_ACCOUNT, "0x12345678", entryNumber + 1); // eslint-disable-line no-await-in-loop
```

Comment:
This iteration got me confused. I _think_ what you're doing here is starting from the first available `entryNumber` and continuing until there are `12` but if so why not iterate from `entryNumber` to `12` instead of `1` to `12`?

Reply:
`entryNumber` relates to which 'set' of `10**15` CLNY are used to submit an entry. The twelve entires might involve sets 1, 5, 13, 300, 341, 444, 510, 630, 635, 687, 731, 899. You are right that we want twelve to be submitted, but the entryNumber has (almost) no relation to the `i` being counted here (which is the number of the submission that is about to be made).

Again, perhaps you're being tripped up by my bad naming here 😢